client/state/sharing/keyring/actions.js

```diff
@@ -12,9 +17,11 @@ import {
 /**
  * Triggers a network request for a user's connected services.
  *
+ * @param {Function} successCallback Function that is called after a successful
+ *                                   request. Default: _.identity.
  * @return {Function} Action thunk
  */
-export function requestKeyringConnections() {
+export function requestKeyringConnections( successCallback = identity ) {
```

Comment:
See https://github.com/Automattic/wp-calypso/pull/8991/commits/9f6a407827bacba84555bee1f0b8ca8e37af7a00

When creating a connection for the first time we need to refresh the keyring connections to be able to show an account chooser. If we don't wait until that request finished we'll get an error that there is no connectable account.